lib/identity_cache/query_api.rb

```diff
@@ -468,7 +485,21 @@ def prefetch_one_association(association, records)
             prefetch_embedded_association(records, association, details)
```

Comment:
I'm curious about why this line is inside the `if details[:embed] == true` block for `cached_has_ones` but before the `if details[:embed] == true` for `cached_has_manys`. Which is correct, and if they're both correct, why does it need to be different in each case?

Reply:
Both seem to work, and I have no idea why. I'll make a note to investigate this in another PR.

Reply:
I think that might matter in the cases where we are calling `prefetch_associations` on already loaded records (i.e. not through the `includes` `fetch_multi` option).  If that is done on records that wasn't loaded from the cache, then it will refetch the record to use its embedded associations, which would include embedded ids.  So by doing this conditionally, `record.send(details[:cached_id_name])` will load the data from the database.

We should probably be consistent here and make this unconditional as well.